pkg/controllers/reconcile.go

```diff
@@ -0,0 +1,97 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	modeljobsv1alpha1 "github.com/caicloud/temp-model-registry/pkg/api/v1alpha1"
+)
+
+func (r *ModelJobReconciler) reconcile(modeljob *modeljobsv1alpha1.ModelJob) (ctrl.Result, error) {
+
+	// Update state if is deleting
+	if !modeljob.ObjectMeta.DeletionTimestamp.IsZero() && modeljob.Status.Phase != modeljobsv1alpha1.ModelJobDeleting {
+		modeljob.Status.Phase = modeljobsv1alpha1.ModelJobDeleting
+		err := r.Status().Update(context.Background(), modeljob)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	// Get a local copy of modeljob's instance.
+	oldModelJob := modeljob.DeepCopy()
+
+	r.reconcileWorkerPod(modeljob)
+
+	// Update modeljob's status.
+	if !equality.Semantic.DeepEqual(modeljob.Status, oldModelJob.Status) {
+		if err := r.Status().Update(context.Background(), modeljob); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ModelJobReconciler) reconcileWorkerPod(modeljob *modeljobsv1alpha1.ModelJob) error {
+	var err error
+
+	pod := &corev1.Pod{}
```

Comment:
Should we use pod directly or job? Maybe we can open an issue to discuss it.

Reply:
I think job is more reasonable too.

Reply:
We can merge it first and open an issue to keep track